### DISCUSSION TITLE
get an error on Style for MetroWindow

### DISCUSSION THREAD

**Discussion by msavazzi** [MahApps/MahApps.Metro#4038] [Q&A]
I just created the project and I get this error on the ShellWindow:
Error	XDG0062	Can only base on a Style with target type that is base type 'Window'.	

here is the window.xaml

<controls:MetroWindow
    x:Class="TitoDoc2020.Views.ShellWindow"
    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
    xmlns:Dialog="clr-namespace:MahApps.Metro.Controls.Dialogs;assembly=MahApps.Metro"
    xmlns:behaviors="clr-namespace:TitoDoc2020.Behaviors"
    xmlns:controls="http://metro.mahapps.com/winfx/xaml/controls"
    xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
    xmlns:i="http://schemas.microsoft.com/xaml/behaviors"
    xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
    xmlns:properties="clr-namespace:TitoDoc2020.Properties"
    xmlns:templateSelectors="clr-namespace:TitoDoc2020.TemplateSelectors"
    Title="{x:Static properties:Resources.AppDisplayName}"
    d:DesignHeight="400"
    d:DesignWidth="600"
    DataContext="{Binding ShellViewModel, Source={StaticResource Locator}}"
    Dialog:DialogParticipation.Register="{Binding}"
    SaveWindowPosition="True"
    **Style="{StaticResource CustomMetroWindow}"**
    WindowStartupLocation="CenterScreen"
    mc:Ignorable="d">

and here is the style in MetroWindow.xaml
<ResourceDictionary
    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
    xmlns:controls="http://metro.mahapps.com/winfx/xaml/controls">

    <Style
        x:Key="CustomMetroWindow"
        BasedOn="{StaticResource {x:Type controls:MetroWindow}}"
        TargetType="{x:Type controls:MetroWindow}">
        <Setter Property="ShowTitleBar" Value="True" />
        <Setter Property="Height" Value="450" />
        <Setter Property="Width" Value="800" />
    </Style>
</ResourceDictionary>

What am I doing wrong?

**Comment by timunie** [comment]
Hi @msavazzi 

you can try the following: 
1. Did you define the right base class in your code behind? 
2. Make sure your nuget packages are restored and up to data https://docs.microsoft.com/en-us/nuget/consume-packages/package-restore 
3. Clean the solution (also delete `bin` and `obj` folder) and rebuild the solution

if the above does not help please make a short sample App, create a new repro on github and share the link here

please also provide information about:
Windows version you use
Visual Studio version you use
MahApps version you use

Happy coding 
Tim

**Reply by msavazzi** [reply]
@timunie if you want I can share the whole project :) so you can see the details

**Reply by timunie** [reply]
Yes please do.

**Reply by timunie** [reply]
Mhh maybe you should create a private repro on GitHub and invite me and @punker76 to it instead of uploading it in a public cloud if it is that confidental. Everybody has now access to it.

**Reply by msavazzi** [reply]
ops... removed share and link :) no rogue access so far :) apparently. Thank you for your smart suggestion. I've create a private repro and added @timunie and @punker76 at it. You're very kind. Thank you!
As is my first try let me know if I did everything ok and you can access

**Reply by timunie** [reply]
Your App runs on my machine without any issue. So I have no idea what is wrong on your side. 

You may try the following:
- Update VisualStudio to the latest version
- delete the folder `obj, `bin`and `.vs`

Happy coding
Tim

**Reply by msavazzi** [reply]
Thank you @timunie
